linuxPackages_latest.prl-tools: 19.4.1-54985 -> 20.0.0-55653

codgician opened the pull request:
## Description of changes

https://kb.parallels.com/en/130212

/cc @wegank for review.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

codgician committed:
linuxPackages_latest.prl-tools: 19.4.1-54985 -> 20.0.0-55653

wegank commented:
Thanks! I've tested on my machine, and it seems like the following items in `nixos/modules/virtualisation/parallels-guest.nix` become unneeded after migrating to FUSE. Would you please drop them in the same PR?

- The option `hardware.parallels.autoMountShares`
- The service `systemd.services.prlfsmountd` (`prlfsmountd` no longer stays in the background, and the service fails anyway, according to `systemctl status prlfsmountd.service`)

codgician committed:
parallels-guest: remove autoMountShares

codgician commented:
> Thanks! I've tested on my machine, and it seems like the following items in `nixos/modules/virtualisation/parallels-guest.nix` become unneeded after migrating to FUSE. Would you please drop them in the same PR?
> 
> * The option `hardware.parallels.autoMountShares` (which is now always true)
> * The service `systemd.services.prlfsmountd` (`prlfsmountd` no longer stays in the background, and the service fails anyway, according to `systemctl status prlfsmountd.service`)

Thanks. I have made the following updates to `parallels-guest`:

- Removed option `autoMountShares` and service definition of `prlfsmountd`.
- Added backward compatibility for removed options.
